File: libparabolic/src/events/downloadcompletedeventargs.cpp
#include "events/downloadcompletedeventargs.h"

using namespace Nickvision::TubeConverter::Shared::Models;

namespace Nickvision::TubeConverter::Shared::Events
{
    DownloadCompletedEventArgs::DownloadCompletedEventArgs(int id, DownloadStatus status, const std::filesystem::path& path, bool showNotification)
        : m_id{ id },
        m_status{ status },
        m_path{ path },
        m_showNotification{ showNotification }
    {
        
    }

    int DownloadCompletedEventArgs::getId() const
    {
        return m_id;
    }

    DownloadStatus DownloadCompletedEventArgs::getStatus() const
    {
        return m_status;
    }

    const std::filesystem::path& DownloadCompletedEventArgs::getPath() const
    {
        return m_path;
    }

    bool DownloadCompletedEventArgs::getShowNotification() const
    {
        return m_showNotification;
    }
}

File: Nickvision.Parabolic.Shared/Events/DownloadCompletedEventArgs.cs
using Nickvision.Parabolic.Shared.Models;
using System;

namespace Nickvision.Parabolic.Shared.Events;

public class DownloadCompletedEventArgs : DownloadEventArgs
{
    public DownloadStatus Status { get; }
    public string Path { get; }
    public ReadOnlyMemory<char> Log { get; }
    public bool ShowNotification { get; }

    public DownloadCompletedEventArgs(int id, DownloadStatus status, string path, ReadOnlyMemory<char> log, bool showNotification) : base(id)
    {
        Status = status;
        Path = path;
        Log = log;
        ShowNotification = showNotification;
    }
}
